### ISSUE TITLE
Add BanNode

### ISSUE THREAD

**Issue by samreid** [phetsims/scenery-phet#548]
For https://github.com/phetsims/circuit-construction-kit-common/issues/541#issuecomment-547178467 we would like a circle-with-a-line-through-it "ban" icon.

Also marking for https://github.com/phetsims/circuit-construction-kit-common/issues/528

Font awesome has an icon for this, but it is not customizable, see https://fontawesome.com/icons/ban

**Comment by samreid** [comment]
I asked on slack:

> Sam Reid 6:32 PM
> Do we already have a kite shape or scenery node for a circle with a line through it?

I thought we had something like that, but couldn't find something after a quick skim.  I created BanNode.js in scenery-phet, and used it in CCK.  

@ariel-phet can you please schedule a reviewer?  It's a short file (53 lines).  Some questions for the reviewer are marked as "TODO" in the file.  Other questions for the reviewer:

* does this component warrant sample usage in the scenery-phet component demo?
* can you find other occurrences of this icon pattern across our repos?

**Comment by ariel-phet** [comment]
@samreid before you do this...I know function builder has such a symbol. If you for instance put in the Warhol Function (the soup can)on the first screen, drag an item through and then try to drag it in reverse you will see this "ban" symbol. We probably don't want to duplicate, so can we first see if that node already exists there? If so, @pixelzoom would probably be the proper reviewer.

**Comment by ariel-phet** [comment]
@pixelzoom in Function Builder is that just an image, or something you drew in code?

**Comment by pixelzoom** [comment]
The symbol is the universal "no" symbol, see https://en.wikipedia.org/wiki/No_symbol. It's known by a number of different emoji names, include "banned", see https://emojipedia.org/no-entry-sign/.  

In Function Builder, the symbol is created programmatically by combining a `SCENERY/Circle` and a `SCENERY/Line`, see `NotInvertibleSymbolNode`. It could also possibly be implemented as a FontawsomeNode (if they have an icon for this symbol) or Unicode (if web-safe fonts support the emoji).

The symbol is displayed when you attempt to drag a card backwards through a function that is non-invertible. For example, follow these steps and see screenshots below:

1. start sim
2. go to Patterns screen
3. drag butterfly card from input carousel, and through builder
4. drag grayscale function (B&W camera icon) to builder (see first screenshot below)
5. attempt to drag butterfly card from output carousel, and backwards through builder (see second screenshot below)

<img width="1075" alt="screenshot_1654" src="https://user-images.githubusercontent.com/3046552/67905597-85519480-fb37-11e9-8139-606eca115d10.png">
<img width="883" alt="screenshot_1656" src="https://user-images.githubusercontent.com/3046552/67905758-06109080-fb38-11e9-83cd-d4bda3778d67.png">

**Comment by pixelzoom** [comment]
Btw... If you want to move `NotInvertibleSymbolNode` to common code, go for it.  It has no dependencies on function-builder (other than namespace).  It should probably be named something more general, like `BannedNode`.

**Comment by samreid** [comment]
I renamed `NotInvertibleSymbolNode` to `BanNode` and migrated it to scenery-phet. I'm also using this code in CCK-AC, you can see it when you click on a capacitor or inductor:

![image](https://user-images.githubusercontent.com/679486/72364657-199db300-36b4-11ea-9a07-51ef8b4eed54.png)

I've also written up several TODOs marked with this issue URL, which would be to discuss during the review.  @pixelzoom are you available to review this code?  For the timeline, it would be nice to get this wrapped up so we can publish it with a draft in a couple of weeks and an RC in 1-2 months.

**Comment by pixelzoom** [comment]
I review the migrated code, looks fine.  

Looks like the only thing that I need to comment on are these TODO items:

```js
  // TODO: What is the best name for this node?  We discussed BannedNode and BanNode, but those terms aren't mentioned
  // TODO: at all in the wikipedia entry, which uses terms like:

As the comment at the top of the files says, "banned" is the emoji name.

  // * prohibition sign
  // * no symbol
  // * no sign
  // * circle-backslash symbol
  // * nay
  // * interdictory circle
  // * universal no
  // TODO: The term "prohibited" is featured prominently in both https://en.wikipedia.org/wiki/No_symbol and
  // TODO: https://emojipedia.org/no-entry-sign/, so I recommend we use a variant of that, such as ProhibitedNode
  // TODO: Or ProhibitedSymbolNode or ProhibitionSignNode, see https://github.com/phetsims/scenery-phet/issues/548
```

Wikipedia and ISO 7001 say: "The general prohibition sign (official name, according to ISO 7010[1]), also known as a no symbol, no sign, circle-backslash symbol, nay, interdictory circle or universal no, ..."    The world doesn't need yet-another variation like `ProhibitedNode ` or `ProhibitedSymbolNode`. `ProhibitionSignNode`, `NoSymbolNode`, `NoSignNode`, or `BannedNode` would all be fine.  My preference would be either `NoSymbolNode` or `BannedNode`, since that's how I've typically heard this symbol referred to (depending on whether the audience is familiar with the ISO or emoji symbol). 

```js
44 fill: 'white' // TODO: Should the default fill be null? See https://github.com/phetsims/scenery-phet/issues/548
```

Yes.

```js
  // TODO: Should we show the circleNode in front of the slashNode to avoid the possibility of "seams"? See https://github.com/phetsims/scenery-phet/issues/548
```

Sounds like a good idea.

**Comment by pixelzoom** [comment]
In reviewing the changes to function-builder, I noticed that the doc for `NotInvertibleSymbolNode` was changed. Important information was lost and the new doc was almost meaningless.  I'm not sure why it was necessary to touch this doc, but I've restored it in the above commit.

**Comment by samreid** [comment]
The above changes do the following:
* Rename BanNode => BannedNode
* Change default fill to null.  I chose a simplistic way of dealing with this in function builder since there is only one occurrence of `new NotInvertibleSymbolNode`, but I thought I'd leave it up to you if you want to move the fill into the declaration of NotInvertibleSymbolNode or leave it at the call site.
* I tried layering the circle in front, but found that is not compatible with a filled circle, so I removed that todo and left the layering alone.

Back to you @pixelzoom
